### DISCUSSION TITLE
How to change the curl that brew uses?

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#1752] [Everyday usage] [answered]
So, I've installed `curl` with brew, and I'd like to use that over the current `curl`, which is the default system one that it uses.

`brew config` shows this:

```
...
Curl: 7.54.0 => /usr/bin/curl
...
```

How do I make it use the newer & better `curl` that it installed? The `/usr/bin/curl` fails to install a package (`theora`), so I have to do a really hacky thing where I verbose & get the command that tries to install `theora` using the old `curl`.

I have to change it from:
` /usr/bin/curl --disable --globoff --show-error `

to:
` /usr/local/Cellar/curl/7.77.0/bin/curl --disable --globoff --show-error `

**Comment by alebcay** [comment, marked as answer]
Setting `HOMEBREW_FORCE_BREWED_CURL` should make Homebrew use brewed `curl` instead of system `curl`.

> `HOMEBREW_FORCE_BREWED_CURL`
> If set, always use a Homebrew-installed curl(1) rather than the system version. Automatically set if the system version of curl is too old.

**Reply by emh789** [reply]
I was just now having the same problem installing `libvorbis`, and this solution worked for me.

`HOMEBREW_FORCE_BREWED_CURL=1 brew install libvorbis`

or put in your .bash_profile (or .bashrc):

`export HOMEBREW_FORCE_BREWED_CURL=1`

**Reply by martianmartian** [reply]
nice one. how to do the same to make brew use 'brew installed ruby' ?

**Reply by gromgit** [reply]
> make brew use 'brew installed ruby'

AFAIK, there's no option for that. Homebrew will automagically vendor Ruby 2.6.8 for your platform, and you can set `HOMEBREW_FORCE_VENDOR_RUBY` in the same way as above to force Homebrew to always use that vendored Ruby.

Also, if a `brew upgrade` upgrades `ruby` _while `brew` is still using it_, the results are unpredictable.

What's your use case for forcing `brew` to use `brew`ed Ruby?

**Comment by 2246sam20** [comment]
I was just facing the same issue while installing the efl using brew install efl  and this solution worked for me too
`HOMEBREW_FORCE_BREWED_CURL=1 brew install efl`
i was getting this error when using the brew install efl

```
==> Downloading https://downloads.xiph.org/releases/theora/libtheora-1.1.1.tar.bz2
curl: (35) error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version
Error: Failed to download resource "theora"
```
